### PR TITLE
Sync SP from FR

### PR DESCRIPTION
**Pull request by bradleybeddoes**

### CODE REVIEW

File: app/jobs/concerns/store_federation_registry_data.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ def update_by_fr_id(dataset, fr_id, attrs)
   end
 
   def record_fr_id(object, fr_id)
+    ds = FederationRegistryObject
+         .where[internal_class_name: object.class.name, fr_id: fr_id]
```
Comment (smangelsdorf) [inline app/jobs/concerns/store_federation_registry_data.rb]: `ModelClass.where[conds]` strikes me as a bit odd. We've had this discussion before, however. Do you recall why this was done?

Comment (bradleybeddoes) [inline app/jobs/concerns/store_federation_registry_data.rb]: It is an alias for `#first` that requires arguments. I don't have any issue with changing implementation if it helps readability.

Comment (bradleybeddoes) [inline app/jobs/concerns/store_federation_registry_data.rb]: e.g. `ds = FederationRegistryObject.first(internal_class_name: object.class.name, fr_id: fr_id)`

Comment (smangelsdorf) [inline app/jobs/concerns/store_federation_registry_data.rb]: I'm talking specifically about the combination of `where` and `[]`... I thought `ModelClass[conds]` was sufficient?

Comment (bradleybeddoes) [inline app/jobs/concerns/store_federation_registry_data.rb]: My understanding is that `ModelClass[]` is for primary key, e.g. `ModelClass[1]`

http://sequel.jeremyevans.net/rdoc/files/doc/querying_rdoc.html#label-Using+a+Primary+Key+-5BSequel-3A-3AModel-5D

Comment (smangelsdorf) [inline app/jobs/concerns/store_federation_registry_data.rb]: It surprised me too when I learned about this, but you're essentially relying on the same behaviour:

```ruby
[2] pry(main)> EntityDescriptor[created_at: '2014-08-13 20:15:00']
=> #<EntityDescriptor @values={:id=>618, :organization_id=>154, :extensions=>nil, :created_at=>2014-08-13 20:15:00 UTC, :updated_at=>2015-11-26 23:37:46 UTC, :known_entity_id=>618, :enabled=>false}>
```

My understanding is that `.where[]` is equivalent to:

```ruby
[3] pry(main)> EntityDescriptor.where({})[created_at: '2014-08-13 20:15:00']
=> #<EntityDescriptor @values={:id=>618, :organization_id=>154, :extensions=>nil, :created_at=>2014-08-13 20:15:00 UTC, :updated_at=>2015-11-26 23:37:46 UTC, :known_entity_id=>618, :enabled=>false}>
```

**Pull request merged**